icicle/include/icicle/fields/field.h

```diff
@@ -44,7 +50,11 @@ class Field
 
   static constexpr HOST_DEVICE_INLINE Field zero() { return Field{CONFIG::zero}; }
 
+#ifdef BARRET
   static constexpr HOST_DEVICE_INLINE Field one() { return Field{CONFIG::one}; }
```

Comment:
why do you keep CONFIG::one is Barret in both cases? If you store it as 1*r in the first place you don't need those ifdefs. Also true for the omegas and other values.
Then maybe the weird thing is that r is not in Montgomery maybe.